src/Domain/Provider.php

```diff
@@ -0,0 +1,38 @@
+<?php declare(strict_types = 1);
+
+namespace SandwaveIo\RealtimeRegister\Domain;
+
+final class Provider implements DomainObjectInterface
+{
+    /** @var string */
+    public $name;
+
+    /** @var string[]|null */
+    public $tlds;
+
+    private function __construct(
+        string $name,
+        ?array $tlds
+    ) {
+        $this->name = $name;
+        $this->tlds = $tlds;
+    }
+
+    public static function fromArray(array $json): Provider
+    {
+        return new Provider(
+            $json['name'],
+            $json['tlds'] ?? []
```

Comment:
This should default to null, as this means that the TLDs are not set, instead of there not being any TLDs.

Reply:
Done: https://github.com/sandwave-io/realtimeregister-php/pull/22/commits/05d5efc506d86688ffa8f7830986ba2951e959fd